src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -272,89 +455,49 @@ object SRAM {
       )
     }
 
-    val _out = Wire(new SRAMInterface(size, tpe, numReadPorts, numWritePorts, numReadwritePorts))
+    val _out = Wire(new SRAMInterface(size, tpe, numReadPorts, numWritePorts, numReadwritePorts, isVecMem))
     val mem = SyncReadMem(size, tpe)
 
-    for (i <- 0 until numReadPorts) {
+    for ((clock, i) <- readPortClocks.zipWithIndex) {
       _out.readPorts(i).data := mem.read(_out.readPorts(i).address, _out.readPorts(i).enable, clock)
     }
```

Comment:
```suggestion
    for ((clock, port) <- readPortClocks.zip(_out.readPort)) {
      port.data := mem.read(port.address, port.enable, clock)
    }
```

I should've caught this with the earlier PR, but unless you are writing performance critical code using Arrays, indexing a Seq is code smell. This same change for the write and readwrite ports will make those loops much cleaner.